rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java

```diff
@@ -307,7 +308,20 @@ private void call() {
 
         if (precompiledContract != null) {
             Metric metric = profiler.start(Profiler.PROFILING_TYPE.PRECOMPILED_CONTRACT_INIT);
-            precompiledContract.init(tx, executionBlock, track, blockStore, receiptStore, result.getLogInfoList());
+            ProgramInvoke contractProgramInvoke = this.program == null ? null : this.program.getInvoke();
```

Comment:
I guess `this.program` is always null at this line of code. So we might need to create an instance of `ProgramInvoke` manually - same as it's done in [here](https://github.com/rsksmart/rskj/pull/2190/files#diff-0a85a5d76a914f023fa43c7433989e77d2c4eca44289ef7cfc0c0ffb87e4846aR368-R369)

Reply:
any thoughts on [this](https://github.com/rsksmart/rskj/pull/2190/files#r1404310701) comment?

Reply:
From what I see ProgramInvoke gets its call stack depth value by getting it from program first and the adding one. So manually creating a ProgramInvoke might not give us the correct value. [Here](https://github.com/rsksmart/rskj/blob/a63d4b8b14fb5a21cb02bad886ce8e69b843d3c9/rskj-core/src/main/java/org/ethereum/vm/program/invoke/ProgramInvokeFactoryImpl.java#L201) is where ProgramInvoke gets the call stack depth.